### PR TITLE
add: stardust-server

### PR DESCRIPTION
**Pull request by Owen-sz**

### CODE REVIEW

File: anda/stardust/black-hole/update.rhai [hunk 1]
```diff
@@ -0,0 +1,7 @@
+if filters.contains("nightly") {
+    rpm.global("commit", gh_commit("StardustXR/black-hole"));
+    if rpm.changed() {
+      rpm.release();
+      rpm.global("commit_date", date());
+    }
+  }
```
Comment (madonuko) [inline anda/stardust/black-hole/update.rhai]:
```suggestion
  }
}
```

Indentation?

File: anda/stardust/black-hole/stardust-black-hole.spec [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Name:           stardust-black-hole
+Version:        %commit_date.%shortcommit
+Release:        1%?dist
+Summary:        Spatial storage for stardust xr.
+URL:            https://github.com/StardustXR/black-hole
+Source0:        https://github.com/StardustXR/black-hole/archive/%commit/black-hole-%commit.tar.gz
+License:        MIT
+BuildRequires:  cargo cmake anda-srpm-macros cargo-rpm-macros mold
+Requires:       libgcc glibc glibm
```
Comment (madonuko) [inline anda/stardust/black-hole/stardust-black-hole.spec]:
```suggestion
```
Automatically added by rpmbuild

File: anda/stardust/black-hole/stardust-black-hole.spec [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+# Exclude input files from mangling
+%global __brp_mangle_shebangs_exclude_from ^/usr/src/.*$
+
+Name:           stardust-black-hole
+Version:        %commit_date.%shortcommit
+Release:        1%?dist
+Summary:        Spatial storage for stardust xr.
+URL:            https://github.com/StardustXR/black-hole
+Source0:        https://github.com/StardustXR/black-hole/archive/%commit/black-hole-%commit.tar.gz
```
Comment (madonuko) [inline anda/stardust/black-hole/stardust-black-hole.spec]:
```suggestion
Source0:        %url/archive/%commit/black-hole-%commit.tar.gz
```
don't repeat yourself

File: anda/stardust/black-hole/stardust-black-hole.spec [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+%description
+Spatial storage for stardust xr
+
+%prep
+%autosetup -n black-hole-%commit
+%cargo_prep_online
+
+%build
+%cargo_build
```
Comment (madonuko) [inline anda/stardust/black-hole/stardust-black-hole.spec]:
```suggestion
```
Don't build twice

**Review by madonuko — CHANGES_REQUESTED** [review]

**Comment by madonuko** [comment]
Also you don't need the if statement in the rhai script

File: anda/stardust/black-hole/update.rhai [hunk 5]
```diff
@@ -4,4 +4,5 @@ if filters.contains("nightly") {
       rpm.release();
       rpm.global("commit_date", date());
     }
-  }
\ No newline at end of file
+  }
```
Comment (madonuko) [inline anda/stardust/black-hole/update.rhai]: mismatch

File: anda/stardust/black-hole/update.rhai [hunk 6]
```diff
@@ -4,4 +4,5 @@ if filters.contains("nightly") {
...
+}
```
Comment (madonuko) [inline anda/stardust/black-hole/update.rhai]: and you don't need the if statement

**Review by madonuko — CHANGES_REQUESTED** [review]

File: anda/stardust/black-hole/update.rhai (hunk 5 above)
Comment (Owen-sz) [reply anda/stardust/black-hole/update.rhai]: Oops

Comment (Owen-sz) [reply anda/stardust/black-hole/update.rhai]: Will fix filenames tonight

File: anda/stardust/black-hole/update.rhai [hunk 7]
```diff
@@ -0,0 +1 @@
+rpm.global("commit", gh_commit("StardustXR/server"));
```
Comment (madonuko) [inline anda/stardust/black-hole/update.rhai]:
```suggestion
rpm.global("commit", gh_commit("StardustXR/server"));
if rpm.changed() {
  rpm.release();
  rpm.global("commit_date", date());
}
```

File: anda/stardust/server/stardust-server.spec [hunk 8]
```diff
@@ -0,0 +1,42 @@
+Name:           stardust-server
+Version:        0.44.1
+Release:        1%?dist
+Summary:        Usable Linux display server that reinvents human-computer interaction for all kinds of XR.
+URL:            https://github.com/StardustXR/server
+Source0:        %url/archive/refs/tags/0.44.1.tar.gz
```
Comment (madonuko) [inline anda/stardust/server/stardust-server.spec]:
```suggestion
Source0:        %url/archive/refs/tags/%version.tar.gz
```

**Review by madonuko — DISMISSED** [review]

**Review by madonuko — DISMISSED** [review]

**Review by madonuko — DISMISSED** [review]

**Review by madonuko — DISMISSED** [review]

**Comment by madonuko** [comment]
Next time please mark this as ready for review (un-draft this)

**Comment by Owen-sz** [comment]
> Next time please mark this as ready for review (un-draft this)

Was waiting for the RPM to be tested before I did that, it used to build before but still didn't preform as expected.

**Review by nothingneko — APPROVED** [review]

**Comment by raboneko** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|f39|[](https://github.com/terrapkg/packages/pull/2305)|
|✅|f40|[](https://github.com/terrapkg/packages/pull/2306)|
|✅|f41|[](https://github.com/terrapkg/packages/pull/2307)|
|✅|el10|[](https://github.com/terrapkg/packages/pull/2308)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/11482481965) for details

**Pull request merged**